Update documentation related to data pull

lekoenig opened the pull request:
This PR cleans up comments and documentation and serves to wrap up the first phase of the pipeline that is focused on the data pulls. I tried to resist the urge to fill the README with everything I might include in a blog post, but I've included additional notes in the README that describe some of the decision making that went into building this pipeline. 

Any feedback is welcome for making sure that all targets and functions are sufficiently documented and their usage clear! 

In particular, I'm considering moving the implementation and documentation around `last_forced_build` (in `_targets.R`) to the README as a way to customize the pipeline without adding unnecessarily complexity to our implementation. Lindsay and I had some back and forth on this in #62, but let me know if you have any thoughts on that based on your review here. Thanks in advance! 

Closes #22

padilla410 commented:
That's perfect for me. I can give this a good chunk of time early next week.

padilla410 commented on 1_inventory/src/get_wqp_inventory.R:
```diff
@@ -63,7 +63,7 @@ inventory_wqp <- function(grid, char_names, wqp_args = NULL, max_tries = 3){
 
 #' Function to transform WQP sites to use a consistent coordinate reference system
 #'
-#' @param sites data frame containing sites returned from the WQP query; contains
+#' @param sites data frame containing sites returned from the WQP query; must contain
 #' columns lon, lat, and HorizontalCoordinateReferenceSystemDatumName.
 #' @param crs_out character string indicating the desired output coordinate reference
 #' system (crs); options include "NAD83" or "WGS84", defaults to "WGS84"
```
The method used to create the doc is slightly inconsistent across functions. See this doc that I've highlighted and also check out [this](https://github.com/lekoenig/ds-pipelines-targets-example-wqp/blob/9252f2c7b779e572c1be5ccdacc507782079972d/1_inventory/src/get_wqp_inventory.R#L1-L19) for a comparison.

I don't have a strong opinion on which way is best overall. the main things that I think are necessary are definitely present in both version (e.g., all arguments listed with brief descriptions and info on data type).

lekoenig replied:
Are you referring to the fact that `@description` or `@value` is not included in all functions? If so, I agree that I wasn't too strict in how I set up the documentation. Rather, I primarily focused on making sure the `@param` fields were complete, and then added other fields as I saw fit. Do you think I should attempt to make the documentation 'structure' more consistent?

lekoenig replied:
I've tried to clean up the documentation to use a consistent style across all functions.  Let me know if you have any further suggestions for improving the doc style/readability.

padilla410 commented on 1_inventory/src/check_characteristics.R:
```diff
@@ -96,7 +97,8 @@ find_similar_characteristics <- function(characteristics_select, param_groups_se
     
     header_selected <- "The following characteristics are already included in the WQP codes cfg file: "
-    header_not_selected <- "The following characteristics might be relevant to the selected parameter and are not included in the WQP codes cfg file: "
+    header_not_selected <- paste0("The following characteristics might be relevant to the selected ",
+                                  "parameter and are not included in the WQP codes cfg file: ")
```
This is such a cool feature of this pipeline

lekoenig replied:
Thanks! 😃  This feature was motivated by questions that came up in DRB IWAAs that boiled down to, _How can I know if I've captured all the different characteristic names?_

padilla410 reviewed (changes requested):
Lauren,

This is a lot of great work. I reviewed all changed files and sprinkled in line-specific comments throughout. Specifically, you've hit a nice balance of including thorough doc without being too text heavy.

As for your question about including info on `force_build` into the `README` - I think the readme needs to cover the basics that someone would want to know about getting up and running on a custom pipeline build. That means it would need to cover 4 topics:
* Changing the area of interest  (done)
* Changing the parameter list
* Changing the dates
* Forcing a rebuild to better handle WQP data upload time lags

Even though it is already covered in the internal doc, I think it should also sit at the top level of the repo.

I would say that most of my edits are optional because you've done a nice job with the doc overall.

I only ran into one problem when running the pipeline. It's documented in the in-line comments and occurred when swapping the AOI method for the shapefile method. I would like to successfully troubleshoot that before we merge this in.

lekoenig commented on README.md:
```diff
@@ -1,12 +1,26 @@
+### Defining the spatial area of interest
+In this workflow we define our spatial area of interest from a set of coordinates that we define in the `_targets.R` file. These coordinates represent the vertices of a simple triangle polygon that covers the spatial extent of our WQP data pull. 
 
 We could also use existing watershed boundaries or another polygon from an external data source to define our area of interest. For example, we could replace the targets `p1_AOI` and `p1_AOI_sf` with targets that download and read in an external shapefile:
```
Fantastic idea! I also learned something new here about how to link to/highlight specific lines ✨

padilla410 commented on 1_inventory/src/get_wqp_inventory.R:
```diff
@@ -76,6 +76,10 @@ transform_site_locations <- function(sites, crs_out = "WGS84"){
   
```
Verified

lekoenig commented:
Thanks again for these suggestions, @padilla410! I think I've addressed all of your comments and requested changes, including making the function documentation internally consistent and adding text to the top-level README that describes how to customize the following parts of the pipeline:

- area of interest
- parameter list
- date range
- forced rebuilds

I opted to simplify the pipeline a bit by moving `last_forced_build` from the main pipeline code to an example in the README documentation. We had discussed and approved this option in #62, so this is just an FYI for @lindsayplatt.

padilla410 commented:
OK! I can't say enough good things about this PR. Well done!
